strength_percentiles.py

```diff
@@ -4,89 +4,130 @@
+    equipment = get_data_from_table(row,EQUIPMENT_COLUMN) # ...
+    squat = get_data_from_table(row,SQUAT_COLUMN) # float
+    bench = get_data_from_table(row,BENCH_COLUMN) # ...
+    deadlift = get_data_from_table(row,DEADLIFT_COLUMN) # ...
+
+    if squat and bench and deadlift: # Valid entries for each lift in row
+        total = squat + bench + deadlift
     else:
-        # Extract lifts from appropriate column in desired format, store
-        gender = contents[gender_column].string
-        professional_status = contents[professional_status_column].string
-        equipment = contents[equipment_column].string
-        try:
-            squat = float(contents[squat_column].string)
-        # No entry for this lift or 'DNF', meaning competitor missed all 3 attempts
-        except (TypeError, ValueError):
-            squat = None
-
-        try:
-            bench = float(contents[bench_column].string)
-        except (TypeError, ValueError):
-            bench = None
-
+        total = None
+
+    result = (squat, bench, deadlift, total)
+    categories = (gender, professional_status, equipment)
+    categorize_and_store_result(categorized_results,result,categories)
+
+def get_data_from_table(row, column):
+    """The input is a row containing all of the information about one lifter's
+    performance at the meet, and the column at which the desired data is
+    located.  This function returns a string if the requested column contains
+    gender, professional status, or equipment data, and returns a float if the
+    requested column contains squat, bench, or deadlift data."""
+    data = row[column].string
+    if column in (SQUAT_COLUMN, BENCH_COLUMN, DEADLIFT_COLUMN):
+        # If the entry at the specified row and column is empty, data will
+        # have value None and TypeError will be thrown.  If the entry is DNF,
+        # meaning the lifter missed all their attempts at the lift, data will
+        # have the value 'DNF' which can't be converted to a float, and
+        # ValueError will be thrown.
         try:
-            deadlift = float(contents[deadlift_column].string)
+            data = float(data)
         except (TypeError, ValueError):
-            deadlift = None
-
-        if squat and bench and deadlift:
-            total = squat + bench + deadlift
-        else:
-            total = None
-
-        numbers = (squat, bench, deadlift, total)
-        lifts['all'].append(numbers)
-
-        if gender == 'Female':
-            lifts['women'].append(numbers)
-        else:
-            lifts['men'].append(numbers)
-
-        if professional_status == 'AM':
-            lifts['amateur'].append(numbers)
+            return None
         else:
-            lifts['professional'].append(numbers)
+            return data
+    return data
+
+def categorize_and_store_result(container, result, categories):
+    """The input is an iterable of categories containing a gender, professional
```

Comment:
In this docstring, it'd be good to say what container, result, categories are.
As an example, see: https://github.com/scikit-learn/scikit-learn/blob/ab93d65/sklearn/linear_model/base.py#L417
Doesn't have to be formatted in that way but you get the idea.